### ISSUE TITLE
File Upload

### ISSUE THREAD

**Issue by gianluca-valentini** [spotify/apollo#256]
Hi,
is there a way to implement file upload using Apollo Rest API?
What is the correct approach to do it?

**Comment by DanHoughton** [comment]
Hi there,

I implemented a small service using this library that provides an endpoint to upload a file to the server.

During my implementation I noticed there wasn't any native support to parse multipart/form-data payloads. Because of this there is an additional dependency in my example that brings in the Apache Commons FileUpload library.

If you'd like to take a look at the example it can be [found here](https://github.com/DanHoughton/apollo/tree/fileUploadExample/examples/fileServer).

If the maintainers would like I can submit this as a pull request, but it might be best to add native support in the Apollo API for multipart/form-data and update this example to use that instead of the Apache Commons library used in my example above.
